### PR TITLE
Check python environment

### PR DESCRIPTION
**Pull request by jgsogo**
* Check a test is not modifying python dependencies: `protobuf` one was installing `protobuf` libraries which upgraded `six` package to a version not compatible with Conan
 * Fix `protobuf` example: if we need to install Python stuff, use a virtualenv

### CODE REVIEW

File: libraries/protobuf/serialization/build.sh [hunk 1]
```diff
@@ -14,7 +12,7 @@ cmake .. -DCMAKE_BUILD_TYPE=Release
 cmake --build .
 
 bin/sensor
-touch __init__.py
 
-popd
-python main.py
+pip install protobuf
+python ../main.py
+pip uninstall -y protobuf
```
Comment (jgsogo) [inline libraries/protobuf/serialization/build.sh]: I'm not able to make it work with a virtualenv 😢 , so I use the main python environment and uninstall the new package afterward

Comment (uilianries) [reply libraries/protobuf/serialization/build.sh]: protobuf is evil :smiling_imp:

Comment (uilianries) [reply libraries/protobuf/serialization/build.sh]: why don't you remove all python package installed by protobuf as last step? So I think is possible to use with virtualenv

Comment (jgsogo) [reply libraries/protobuf/serialization/build.sh]: the problem with the virtualenv is that I'm not able to install `protobuf` inside the virtualenv in Linux (like I do in Windows or Mac), and to pass this new check (--- tests do not modify the python environment ---) I need to remove it here

Comment (jgsogo) [reply libraries/protobuf/serialization/build.sh]: Windows has failed too 😢 ...so it is something related to the levels of bash calling bash calling python calling bash... so I've moved to the `uninstall` strategy

**Comment by jgsogo** [comment]
I wrote my concerns about this example in https://github.com/conan-io/examples/issues/50

**Pull request merged**